src/inspect_ai/tool/_tools/_computer/_computer.py

```diff
@@ -0,0 +1,200 @@
+            validate_no_args(args, "left_click")
+            return await left_click()
+
+        async def left_click_drag_action(args: str) -> ToolResult:
+            endx, endy = validate_coordinates(args, "left_click_drag")
+            return await left_click_drag(endx, endy)
+
+        async def right_click_action(args: str) -> ToolResult:
+            validate_no_args(args, "right_click")
+            return await right_click()
+
+        async def middle_click_action(args: str) -> ToolResult:
+            validate_no_args(args, "middle_click")
+            return await middle_click()
+
+        async def double_click_action(args: str) -> ToolResult:
+            validate_no_args(args, "double_click")
+            return await double_click()
+
+        async def screenshot_action(args: str) -> ToolResult:
+            validate_no_args(args, "screenshot")
+            return await screenshot()
+
+        # Dictionary to map actions to functions
+        action_map: dict[str, ActionFunction] = {
+            "key": key_action,
+            "type": type_action,
+            "cursor_position": cursor_position_action,
+            "mouse_move": mouse_move_action,
+            "left_click": left_click_action,
+            "left_click_drag": left_click_drag_action,
+            "right_click": right_click_action,
+            "middle_click": middle_click_action,
+            "double_click": double_click_action,
+            "screenshot": screenshot_action,
+        }
+
+        # Execute the corresponding function
+        try:
+            if action in action_map:
+                result: ToolResult | Awaitable[ToolResult] = action_map[action](
+                    arguments
+                )
+                return (await result) if asyncio.iscoroutine(result) else result  # type: ignore
+            else:
+                available_actions = ", ".join(sorted(action_map.keys()))
+                return f"Unknown action: {action}. Available actions are: {available_actions}"
+        except Exception as e:
+            return f"An error occurred while executing the action: {e}"
+
+    def validate_coordinates(args: str, action_name: str) -> tuple[int, int]:
+        """Validate and parse X,Y coordinate arguments."""
+        try:
+            # Strip any extra whitespace and split
+            coords = [x.strip() for x in args.split()]
+
+            if len(coords) != 2:
+                raise ValueError(
+                    f"{action_name} requires exactly 2 integer coordinates (X Y)"
+                )
+
+            # Convert to integers and validate
+            x, y = map(int, coords)
+
+            # Basic sanity check for coordinate values
+            if x < 0 or y < 0:
+                raise ValueError("Coordinates must be non-negative integers")
+
+            return x, y
+        except ValueError as e:
+            raise ValueError(f"Invalid coordinates for {action_name}: {str(e)}")
```

Comment:
I would raise a `ToolParsingError` here as well (same for `validate_type` and `validate_no_args`)